src/earn/hooks.ts

```diff
@@ -93,6 +93,15 @@ export function useAaveRewardsInfoAndPrepareTransactions({
   return { asyncRewardsInfo, asyncPreparedTransactions }
 }
 
+export function useEarnPositionProviderName(providerId: string) {
+  const pools = useSelector(earnPositionsSelector)
+  const providerName = pools.find((pool) => pool.appId === providerId)?.appName
```

Comment:
Is there any case we'd want to use a fallback provider name? But that might add to the confusion when debugging.

Reply:
@MuckT it looks like the provider name is used in the translation string like "Deposited to {{provider}} pool", etc. I guess a default could be "Earn", but then it wouldn't get translated. 

We could add in a fallback subtitle to show if we want because with my implementation here the subtitle is just omitted if providerName isn't found.

Reply:
It's fine to avoid a fallback with the conditional rendering as described.